### PR TITLE
Start onboarding feature

### PR DESCRIPTION
**Pull request by jonahduckworth**
- new admin route
- tables for officials and supervisors
- delete button works
     - confirmation modal opens
          - abstracted `ConfirmationModal` from `SelectedGames`
     - dispatch `deleteUser` on confirm, then: dispatch `getOfficialsList`, close modal, turn off loading, and show success toast

### LOG IN TO AJHL LEAGUE TO TEST
- email: admin@ajhl.ca
- password: password
- delete either of the accounts that are there

Screenshot 2024-01-14 at 7 18 42 PM

- invite user card
     - on click opens modal
     - error handling on empty text field and invalid email
     - on send: dispatch `getOfficialsList`, close modal, turn off loading, and show success toast
          - still need to build invite endpoint

Screenshot 2024-01-15 at 12 40 07 PM

### CODE REVIEW

File: src/components/Calendar/SelectedGames/SelectedGames.tsx [hunk 1]
```diff
@@ -487,63 +488,15 @@ const SelectedGames = ({ onClose }) => {
...
-                          className="inline-flex justify-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-success-400 hover:bg-success-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
-                          onClick={handleDeleteGameClick}
-                        >
-                          Delete
-                        </button>
-                      </div>
-                    </div>
-                  </div>
-                </div>
+                <ConfirmationModal
```
Comment (rymccue) [inline src/components/Calendar/SelectedGames/SelectedGames.tsx:491]: This is clean 👌

File: src/templates/Admin/Admin.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+const Admin: React.FC<any> = () => {
+  const dispatch = useAppDispatch();
+  const loading = useAppSelector((state) => state.officials.loading);
+  const league = useAppSelector((state) => state.user.currentLeague);
+
+  const [isInviteUserModalOpen, setIsInviteUserModalOpen] =
+    useState<boolean>(false);
+  const [isConfirmationModalOpen, setisConfirmationModalOpen] =
+    useState<boolean>(false);
+  const [uid, deleteUid] = useState<string>('');
```
Comment (rymccue) [inline src/templates/Admin/Admin.tsx:29]: This doesn't really delete the uid, does it?

File: src/templates/Admin/components/InviteUserModal.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  const handleSend = async () => {
+    // Check if email is empty
+    if (!email) {
+      setError('Email is required');
+      return;
+    }
+
+    // Check if email is valid
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (rymccue) [inline src/templates/Admin/components/InviteUserModal.tsx:33]: Is this a proper email regex? Looks smaller than I've seen before but might this is just cleaner!

**Review by rymccue — CHANGES_REQUESTED** [review]
Added a few comments, overall this is awesome though!! 🔥

File: src/templates/Admin/Admin.tsx (hunk 2 above)
Comment (jonahduckworth) [reply src/templates/Admin/Admin.tsx:29]: would `const [deleteUid, setDeleteUid] = useState<string>('');` make more sense?

File: src/templates/Admin/components/InviteUserModal.tsx (hunk 3 above)
Comment (jonahduckworth) [reply src/templates/Admin/components/InviteUserModal.tsx:33]: [chatGPT's response](https://chat.openai.com/share/4d882488-4c16-4888-b34e-347dffbf3db7)

tl;dr this regex asserts something@something.something

File: src/templates/Admin/Admin.tsx (hunk 2 above)
Comment (rymccue) [reply src/templates/Admin/Admin.tsx:29]: What about `const [uidToDelete, setUidToDelete] = useState<string>('');`? It explains it a bit more.

File: src/templates/Admin/components/InviteUserModal.tsx (hunk 3 above)
Comment (rymccue) [reply src/templates/Admin/components/InviteUserModal.tsx:33]: Oh nice, yeah and that works with `ryan+test@ryanmccue.ca` which I didn't expect

**Review by rymccue — APPROVED** [review]
